### ISSUE TITLE
BasicBitcoinJsonRPCClient uses hard-coded wallet name

### ISSUE THREAD

**Issue by darwin** [ACINQ/eclair#1538]
Tracked it down to this [line][0]:
https://github.com/ACINQ/eclair/blob/662e0c4bcc1dec24f9aad2bd526866a8a056153f/eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BasicBitcoinJsonRPCClient.scala#L40

### Background story

Today I wanted to fix broken tests in [simverse](https://github.com/darwin/simverse).

Here are my findings (note that I don't follow bitcoin core development that closely, so I may be missing something): 

1. bitcoind used to create a default wallet with empty name. It seems to be no longer the case. Newly I have to first create some wallet via `bitcoin-cli createwallet <name>`. See my commit [here][1].

2. eclair started failing with `"bitcoind must have wallet support enabled"`, I double checked that my bitcoind really had wallet support enabled and started googling the issue. I found the same problem reported [here][2].

3. I looked at the place where the [error is raised][3] and decided to investigate further.

4. I found out that the issue is that eclair has [hardcoded empty wallet name][0].

5. bitcoind logic for selecting wallet for RPC request is implemented [here][4].

### Possible suggestions

1. BasicBitcoinJsonRPCClient should be configurable to allow wallet name selection (not sure how to propagate this configurability further)

2. By default BasicBitcoinJsonRPCClient should have unspecified name (which is different from empty string), If I understand the rpcwallet.cpp correctly, it would be more flexible in this case not to add "/wallet/" into endpoint url. Bitcoin RPC would then pick the only wallet it has loaded regardless of its name (or return an error about multi-wallet scenario via `RPC_WALLET_NOT_SPECIFIED` or an error about no-wallet scenario via `RPC_METHOD_NOT_FOUND` or any other error which could be implemented on bitcoind side in the future). This would be more flexible than assuming default wallet name to be empty string.

3. Secondary problem is that Setup.scala#L150 swallows original RPC error. Even if you didn't resolve points 1 or 2, the code should be updated to include original RPC error. It should not blindly return `"bitcoind must have wallet support enabled"` which may be misleading, because it assumes particular behaviour of bitcoind which may be subject of future changes.

[0]: https://github.com/ACINQ/eclair/blob/662e0c4bcc1dec24f9aad2bd526866a8a056153f/eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BasicBitcoinJsonRPCClient.scala#L40
[1]: https://github.com/darwin/simverse/commit/296805226fd8d249cd5cfabe1a6d20f712cdbd7c
[2]: https://gitter.im/ACINQ/eclair?at=5e48f793c8da1343d4540a44
[3]: https://github.com/ACINQ/eclair/blob/daddfc007fe569c89bb854ef5d672614d397e91e/eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala#L150
[4]: https://github.com/bitcoin/bitcoin/blob/1b313cacc99a1b372238f9036abed5491f9d28f7/src/wallet/rpcwallet.cpp#L102

**Comment by darwin** [comment]
Just found a mention about this new behaviour in Bitcoin Optech:
https://bitcoinops.org/en/newsletters/2020/09/23/#bitcoin-core-15454

Direct link to relevant PR:
https://github.com/bitcoin/bitcoin/pull/15454

**Comment by t-bast** [comment]
Thanks for the detailed issue, I'll have a look at this in more details.

We made some modifications to this in https://github.com/ACINQ/eclair/pull/1334, and during this PR I suggested adding the ability to configure which wallet to use. In the end we didn't merge that code because there were risks of footguns for users, but maybe it's worth considering it again (and accept that users will have to know what they're doing when they change their wallet).

I'll make some tests, but I'm not sure removing the "/wallet" part will work with bitcoind 0.18.1 (and we currently have backwards-compatibility until that version). We'll need to drop backwards-compatibility with older versions of bitcoin core for fee bumping, maybe at that point it will make sense to also drop the "/wallet" suffix.
